### PR TITLE
Add agent field

### PR DESCRIPTION
**Pull request by patrobinson**
## Description

The Agent field provides information about who made the respective log entry.
This PR adds that field to the data we ingest from the [Log Entries API
](https://developer.pagerduty.com/api-reference/c661e065403b5-list-log-entries)

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Related issues

nil

## Migration notes

nil

## Extra info

nil

### CODE REVIEW

**Comment by patrobinson** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by tovbinm** [comment]
recheck

File: sources/pagerduty-source/src/pagerduty.ts [hunk 1]
```diff
@@ -72,6 +72,7 @@ export interface LogEntry extends PagerdutyObject {
   readonly incident: PagerdutyObject;
   readonly service: PagerdutyObject;
   readonly event_details?: Record<string, any>; // e.g. trigger events have "description" detail
+  readonly agent: PagerdutyObject;
```
Comment (tovbinm) [inline sources/pagerduty-source/src/pagerduty.ts:75]: What populates this field? Is it coming from a response from Pagerduty API?

**Review by tovbinm — APPROVED** [review]

**Comment by tovbinm** [comment]
```
FAIL  test/pagerduty.test.ts
  ● Test suite failed to run

    test/pagerduty.test.ts:27:11 - error TS2741: Property 'agent' is missing in type '{ id: string; type: string; summary: string; self: string; html_url: string; created_at: string; incident: { id: string; type: string; summary: string; self: string; html_url: string; }; service: { id: string; type: string; summary: string; self: string; html_url: string; }; }' but required in type 'LogEntry'.

    27     const logEntry: LogEntry = {
                 ~~~~~~~~

      src/pagerduty.ts:79:12
        79   readonly agent: PagerdutyObject;
                      ~~~~~
        'agent' is declared here.
```
https://github.com/faros-ai/airbyte-connectors/actions/runs/5861517768/job/15898371992?pr=1110#step:6:1787

**Comment by catkins** [comment]
Nice one @patrobinson 👏

Is this one good to go @tovbinm?

**Pull request merged**